### PR TITLE
vmm: api: Make VmSendMigrationData.local optional

### LINKED ISSUE #3756: `local` field of `SendMigrationData` should be required in open-api spec
It seems that a new field `local` has been added to `SendMigrationData` in API since recent releases. When calling `send-migration` API without this new field, the server would return an error saying SerdeJsonDeserialize(Error(\"missing field `local`\", line: 1, column: 54)). Due to this behaviour, this field should really be marked as required in the open-api spec (I'm using this open-api spec to generate Golang client-side code, so I'm depending on this doc's accuracy here). Or if this behaviour is not intended, maybe we should then change the server implementation to make it optional and false by default.

Would like to send a PR here, just let me know which way to go.

### PR DESCRIPTION
**Pull request by fengye87**
This should resolve #3756 .

### CODE REVIEW

File: vmm/src/api/mod.rs [hunk 1]
```diff
@@ -196,6 +196,7 @@ pub struct VmSendMigrationData {
     /// URL to migrate the VM to
     pub destination_url: String,
     /// Send memory across socket without copying
+    #[serde(default)]
```
Comment (rbradford) [inline vmm/src/api/mod.rs:199]: Here is a slightly better commit message

```
vmm: api: Make for VmSendDataMigration.local optional

Fixes: #3756
```

Comment (fengye87) [reply vmm/src/api/mod.rs:199]: done

**Comment by sboeuf** [comment]
I think you forgot to update `vmm/src/api/openapi/cloud-hypervisor.yaml` with the following:
```diff
diff --git a/vmm/src/api/openapi/cloud-hypervisor.yaml b/vmm/src/api/openapi/cloud-hypervisor.yaml
index 135028d2..d86f061c 100644
--- a/vmm/src/api/openapi/cloud-hypervisor.yaml
+++ b/vmm/src/api/openapi/cloud-hypervisor.yaml
@@ -1072,6 +1072,8 @@ components:
           type: string
 
     SendMigrationData:
+      required:
+      - destination_url
       type: object
       properties:
         destination_url:
```

**Comment by fengye87** [comment]
@sboeuf Yeah, `destination_url` should be required, though I think it's not related to this issue. BTW, `receiver_url` (of ReceiveMigrationData struct) should be required too. Don't know if there's more such mistakes in the open-api spec, but they're not affecting my auto-generated Golang client-side code at least for now.

**Comment by sboeuf** [comment]
> @sboeuf Yeah, `destination_url` should be required, though I think it's not related to this issue. BTW, `receiver_url` (of ReceiveMigrationData struct) should be required too. Don't know if there's more such mistakes in the open-api spec, but they're not affecting my auto-generated Golang client-side code at least for now.

Even if not affecting the auto-generated Golang code, better to have them fixed when we notice them.

**Pull request merged**